person.rb

```diff
@@ -33,8 +31,8 @@ def correct_name?
     @name
   end
 
-  def add_rental(book, date)
-    rental = Rental.new(date, book, self)
-    @rentals << rental
+  def add_rental(rental)
+    rentals << rental unless rentals.include?(rental)
+    puts "Rental added to person: Person ID: #{id}, Rental Date: #{rental.date}, Book Title: #{rental.book.title}"
```

Comment:
- [ ] If I want to create person. It's giving me `Invalid book title or person name!` issue. Due to this reason, I was not able to test your list of rentals. Kindly fix and make sure your list of rentals working as expected.